suite-native/token-definitions/src/tokenDefinitionsMiddleware.ts

```diff
@@ -0,0 +1,69 @@
+import { createMiddlewareWithExtraDeps } from '@suite-common/redux-utils';
+import { getNetworkFeatures, isEthereumBasedNetwork, networks } from '@suite-common/wallet-config';
+import {
+    accountsActions,
+    getTokenDefinitionThunk,
+    selectSpecificTokenDefinition,
+} from '@suite-common/wallet-core';
+
+export const prepareTokenDefinitionsMiddleware = createMiddlewareWithExtraDeps(
+    (action, { dispatch, next, getState }) => {
+        next(action);
+
+        if (
+            accountsActions.createAccount.match(action) ||
+            accountsActions.updateAccount.match(action)
+        ) {
+            const { symbol } = action.payload;
+
+            const networkFeatures = getNetworkFeatures(symbol);
+
+            if (networkFeatures.includes('token-definitions')) {
+                action.payload.tokens?.forEach(token => {
+                    const contractAddress = token.contract;
+
+                    const tokenDefinition = selectSpecificTokenDefinition(
+                        getState(),
+                        symbol,
+                        contractAddress,
+                    );
+
+                    const network = networks[symbol];
+                    if (
+                        isEthereumBasedNetwork(network) &&
+                        (!tokenDefinition || tokenDefinition.error)
+                    ) {
+                        dispatch(
+                            getTokenDefinitionThunk({
+                                networkSymbol: symbol,
+                                chainId: network.chainId,
+                                contractAddress,
+                            }),
+                        );
+                    }
+                });
+            }
+        }
+
+        // Fetch token definitions for each token of the discovery created account.
+        if (accountsActions.createIndexLabeledAccount.match(action)) {
+            const { tokens, symbol } = action.payload;
+
+            const network = networks[symbol];
+            if (isEthereumBasedNetwork(network)) {
+                const { chainId } = network;
+                tokens?.forEach(token => {
+                    dispatch(
+                        getTokenDefinitionThunk({
+                            networkSymbol: 'eth',
```

Comment:
`isEthereumBasedNetwork` should be true also for `matic` so this might better be

`networkSymbol: 'eth',` -> `networkSymbol: network.networkType`

Reply:
same comment as above applies.

Reply:
If we enable polygon in the future and do not do any changes, other appearances will work the same (only for 'eth' as this is in the condition), but here it will wrongly work with `eth` instead of `matic` due to `isEthereumBasedNetwork`. 

Please either support the network type of network (suggested solution) or change:

`if (isEthereumBasedNetwork(network)) {` to `if (network.symbol === 'eth') {`

Reply:
You are right, good catch, this could be troublesome later when merged 🤯.  fixed by 8aa48593ed